### PR TITLE
store: switch connectivity check to use v2/info

### PR DESCRIPTION
**Pull request by chipaca**
This is the first use of the new v2/info endpoint. It uses abbreviated
structs to save a little bit of time & space in decoding compared to
the full decode that'll be needed for _actual_ info usage.

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -2372,30 +2373,46 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
...
 	//       when we drop "core" in the move to snapd/core18/etc, change this
-	deetsURL := s.endpointURL(path.Join(detailsEndpPath, "core"), url.Values{"fields": {"anon_download_url"}})
-	hosts = append(hosts, deetsURL.Host)
-
-	var remote snapDetails
-	resp, err := httputil.RetryRequest(deetsURL.String(), func() (*http.Response, error) {
-		return s.doRequest(context.TODO(), s.client, &requestOptions{Method: "GET", URL: deetsURL}, nil)
+	infoURL := s.endpointURL(path.Join(snapInfoEndpPath, "core"), url.Values{
+		"fields":       {"download"},
+		"architecture": {"amd64"},
```
Comment (mvo5) [inline store/store.go]: Curious, why hardcode the architecture here?

Comment (chipaca) [reply store/store.go]: I just need to specify one. I could also use the store's, I guess :-)

**Comment by chipaca** [comment]
No need to backport AFAIK.

File: store/store.go (hunk 1 above)
Comment (zyga) [reply store/store.go]: Can you add a note explaining this.

File: store/store.go [hunk 2]
```diff
@@ -2372,30 +2373,48 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
...
 }
 
+// abbreviated info structs just for the download info
+type storeInfoChannelAbbrev struct {
+	Download storeSnapDownload `json:"download"`
+}
+
+type storeInfoAbbrev struct {
+	// discard anything beyond the first entry
+	ChannelMap [1]storeInfoChannelAbbrev `json:"channel-map"`
```
Comment (bboozzoo) [inline store/store.go]: It'd be nice to add a test with more than one object in `channel-map`.

**Pull request merged**